src/Tests/dotnet-new.Tests/TemplateDiscoveryTests.cs

```diff
@@ -0,0 +1,258 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Cli.New.IntegrationTests
+{
+    public class TemplateDiscoveryTests : BaseIntegrationTest
+    {
+        private readonly ITestOutputHelper _log;
+
+        public TemplateDiscoveryTests(ITestOutputHelper log) : base(log)
+        {
+            _log = log;
+        }
+
+        [Fact]
+        public async Task CanRunDiscoveryTool()
+        {
+            string home = CreateTemporaryFolder(folderName: "Home");
+            string testDir = CreateTemporaryFolder();
+            string testTemplatesPpackagePath = PackTestNuGetPackage(_log);
+            using var packageManager = new PackageManager();
+            packageManager.PackageLocation = Path.GetDirectoryName(testTemplatesPpackagePath)!;
+            string packageLocation = await packageManager.GetNuGetPackage("Microsoft.Azure.WebJobs.ProjectTemplates").ConfigureAwait(false);
+            new DotnetNewCommand(
+                _log,
+                "tool-manifest")
+                .WithCustomHive(home)
+                .WithWorkingDirectory(testDir)
+                .Execute()
+                .Should()
+                .Pass();
+            new DotnetToolCommand(
+                _log,
+                "install",
+                "Microsoft.TemplateSearch.TemplateDiscovery",
+                "--version",
+                TemplatePackageVersion.MicrosoftTemplateSearchTemplateDiscoveryPackageVersion)
+                .WithWorkingDirectory(testDir)
+                .Execute()
+                .Should()
+                .Pass();
```

Comment:
those need to be called for other tests as well, if they are uncommented. 
Consider creating fixture which does it once.

Reply:
It's my missing. Created the fixture.